### ISSUE TITLE
Pylance Language Server & Notebooks

### ISSUE THREAD

**Issue by DonJayamanne** [microsoft/vscode-jupyter#5568]
### TLDR: Start multiple langauge servers, one per kernel (each LS is then isolated)

**Discussion from last week:**
* Existing Language server to ignore notebook schemes
* Start a new language server for each kernel (akin to starting language servers for each workspace in Python extension)
	* Ensure the notebook scheme is supported by this LS (this way even if we have 2 language servers, the Python language server will not interfere with the Notebook Language Server)
	* If user opens 2 notebooks pointing to same kernel (same interpreter), then re-use the LS
* No real perf concerns at this stage when starting multiple LS
	*  Though there could be slight delays in getting completions (as LS takes a few seconds to analyse the interpreter)
* Start an LS as soon as a notebook is opened
* Dispose old LS when a kernel is changed & there are no more documents using the interpreter for which we spun up LS

**Questions:**
* Does Jupyter start these LS or Python?
* FYI - Pylance would like to be its own extension & move stuff out of Python extension
